src/Core/src/Core/FontsMauiAppBuilderExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Maui.Hosting;
+using Microsoft.Maui.Hosting.Internal;
+
+namespace Microsoft.Maui
+{
+	public static class FontsMauiAppBuilderExtensions
+	{
+		public static MauiAppBuilder ConfigureFonts(this MauiAppBuilder builder)
```

Comment:
Is the plan to eventually collapse this (and similar ones) down as well?

either to

```C#
appBuilder.Fonts.AddFont
```

or
```C#
appBuilder.AddFont
```

Reply:
I was on the fence about this for no reason than we need some sort of process for identifying what we want to add there. Once it is added, it can't really be removed. Maybe fonts is not bad, but we need a process of ensuring we don't willy-nilly add properties. We can also help people by renaming this type `Forms` 😉 

It is also breaking to change the property type.

Fonts might be good to add, but we need to make sure we got rules!

Reply:
I like appBuilder.Fonts.*

Reply:
How does ASP.NET handle new things? Is it just added to the type as a property? How do you prevent growth for everything? 

We have app lifecycle events, platform effects (which is maybe something that is old and should not be used), fonts, image source handlers, and possibly a few others. We will end up with many things on there. 

I suppose it is not _bad_, but may cause issues later on if we got things on there that have changed and now no longer make sense:

```csharp
class MauiAppBuilder {
    IServiceCollection Services;
    ConfigurationManager Configuration;
    ILoggingBuilder Logging;
    ConfigureHostBuilder Host;
    IFontCollection Fonts;
    IEffectsCollection Effects;
    ILifecycleEventManager LifecycleEvents;
    IImageSources ImageSources;
}
```

Reply:
There are 2 things being created here:
- The builder (wrapper around the host)
- The application itself (this is where you'll see most ASP.NET primitives exposed). These are things in the service provider that 90% of application will try to access top level https://github.com/dotnet/aspnetcore/blob/3c089fa29a441a390f2786e5107f0dcc8397898f/src/DefaultBuilder/src/WebApplication.cs#L40-L65.

The builder exposes `IHostBuilder` and `IWebHostBuilder` property because the existing ecosystem need to be able to configure more advanced things.

Generally, I think you want to use the 80/20 rule while determining what to expose top level.

Reply:
@PureWeen  then maybe Fonts is cool here. I think things like effects and image sources are more for library authors. Fonts and maybe lifecycle events are user app things. Although, lifecycle events are a bit rare I think? Since most people using that will either be people doing native libraries or library authors.

Maybe just fonts?

Reply:
And handlers. I forgot handlers are like essential for anything.